xbmc/guilib/GUIEditControl.cpp

```diff
@@ -746,4 +746,5 @@ void CGUIEditControl::SetFocus(bool focus)
   m_smsTimer.Stop();
   g_Windowing.EnableTextInput(focus);
   CGUIControl::SetFocus(focus);
+  SetInvalid();
```

Comment:
`SetFocus()` itself is not performing the invalidation and should not care about it. It's just doing the focus toggle. Mind elaborating why you think it should be called before?